src/client/LanguageClient.ts

```diff
@@ -88,15 +98,42 @@ async function addSubscriptions(context: ExtensionContext) {
   context.subscriptions.push(
     commands.registerCommand('metaed.deploy', () => {
       (async () => {
+        const deployDirectoryPath = getOdsApiDeploymentDirectory();
+        const implementationFolderPath = path.join(deployDirectoryPath, 'Ed-Fi-ODS-Implementation');
+        const odsFolderPath = path.join(deployDirectoryPath, 'Ed-Fi-ODS');
+        const drivePattern = /[a-zA-Z]:/;
+        const endsWithSlash = /[/\\]$/;
+
         if (!acceptedLicense()) {
           await showErrorNotification(
             'You must first accept the Ed-Fi License Agreement under File -> Preferences -> Settings.',
           );
           return;
         }
 
-        if (getOdsApiDeploymentDirectory() === '') {
-          await showInfoNotification('To deploy, set Ods Api Deployment Directory under File -> Preferences -> Settings.');
+        if (deployDirectoryPath === '') {
+          await showErrorNotification('To deploy, set Ods Api Deployment Directory under File -> Preferences -> Settings.');
+          return;
+        }
+        if (!isValidDirectory(deployDirectoryPath)) {
+          await showErrorNotification(
+            'Directory path not found, set proper Ods Api Deployment Directory under File -> Preferences -> Settings.',
+          );
+          return;
+        }
+        if (process.platform.indexOf('win') !== -1) {
```

Comment:
I think you want "win32" for your check, like `process.platform === 'win32'`

See https://nodejs.org/api/process.html#process_process_platform